common/logisticspipes/gui/GuiGuideBook.java

```diff
@@ -0,0 +1,331 @@
+				page++;
+				prevPageBtn.enabled = true;
+			}
+			if (page == pageMax) nextPageBtn.enabled = false;
+			else nextPageBtn.enabled = true;
+			sliderProgress = 0.0F;
+			updateSlider();
+			text = splitLines(StringUtils.translate("book.quickstart." + page), fontRenderer, maxLength);
+		}
+		if (button.id == 1) {
+			if (page > 1) {
+				page--;
+				nextPageBtn.enabled = true;
+			}
+			if (page == 1) prevPageBtn.enabled = false;
+			else prevPageBtn.enabled = true;
+			sliderProgress = 0.0F;
+			updateSlider();
+			text = splitLines(StringUtils.translate("book.quickstart." + page), fontRenderer, maxLength);
+		}
+	}
+
+	@Override
+	protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
+		if (slider.isMouseOver()) this.dragging = true;
+		super.mouseClicked(mouseX, mouseY, mouseButton);
+	}
+
+	@Override
+	protected void mouseClickMove(int mouseX, int mouseY, int clickedMouseButton, long timeSinceLastClick) {
+		this.mouseX = mouseX;
+		this.mouseY = mouseY;
+		updateSlider();
+		super.mouseClickMove(mouseX, mouseY, clickedMouseButton, timeSinceLastClick);
+	}
+
+	@Override
+	protected void mouseReleased(int mouseX, int mouseY, int state) {
+		this.dragging = false;
+		super.mouseReleased(mouseX, mouseY, state);
+	}
+
+	@Override
+	public void onGuiClosed() {
+		ItemGuideBook.setCurrentPage(page, sliderProgress, hand);
+		super.onGuiClosed();
+	}
+
+	@Override
+	public void onResize(Minecraft mcIn, int w, int h) {
+		// Updating drawing constrains
+		acrossX = w * 6 / 8 - 2 * tileSize;
+		acrossY = h * 6 / 8 - 2 * tileSize;
+		leftX = w * 1 / 8;
+		topY = h * 1 / 8;
+		rightX = (leftX + tileSize) + (acrossX);
+		bottomY = (topY + tileSize) + (acrossY);
+		// Setting slider constraints
+		sliderTopY = topY + borderSize;
+		sliderBotY = bottomY + borderSize - slider.height;
+		slider.x = rightX + borderSize - slider.height;
+		// Setting text constraints
+		maxLength = (int) (acrossX + (3 / (float) 2) * borderSize - slider.width);
+		textX = leftX + 24;
+		textY = topY + 24;
+		maxLines = (acrossY + borderSize) / 10;
+		// Updating text splitting
+		text = splitLines(StringUtils.translate("book.quickstart." + page), fontRenderer, maxLength);
+		super.onResize(mc, w, h);
+	}
+
+	/*
+	 * Make the slider button follow the mouse if the click was performed over it as well as
+	 * calculate sliderProgress based on the slider position between it's max and min Y.
+	 */
+	protected void updateSlider() {
+		if (dragging && draggable) {
+			if (mouseY < sliderTopY + slider.height / 2) {
+				slider.y = sliderTopY;
+				sliderProgress = (slider.y - sliderTopY) / (float) (sliderBotY - sliderTopY);
+			} else if (mouseY > sliderBotY + slider.height / 2) {
+				slider.y = sliderBotY;
+				sliderProgress = (slider.y - sliderTopY) / (float) (sliderBotY - sliderTopY);
+			} else {
+				slider.y = mouseY - slider.height / 2;
+				sliderProgress = (slider.y - sliderTopY) / (float) (sliderBotY - sliderTopY);
```

Comment:
This line can be extracted out of all the if blocks

Reply:
I agree, but this was replaced with the change below :)